html/gui/js/libraries/PlotlyUtilities.js

```diff
@@ -161,10 +213,288 @@ function generateChartOptions(record, params) { // eslint-disable-line no-unused
         }
     };
 
-    var ret = {
-        chartData: data,
-        chartLayout: layout
+    return { data, layout };
+}
+/**
+ * Returns array of axis layout keys for the axis that represents
+ * the range of the data
+ *
+ * @param  {Object} clickEvent - Plotly JS click event object.
+ * @param  {Array} traceDivs - Array of chart series svg elements
+ * @return {Object} point from Plotly JS click event
+ */
+function getClickedPoint(clickEvent, traceDivs) { // eslint-disable-line no-unused-vars
+    if ((traceDivs && traceDivs.length === 0) || (clickEvent.points && clickEvent.points.length === 0)) {
+        return null;
+    }
+    const findPoint = (evt, traces) => {
+        for (let i = 0; i < traces.length; i++) {
+            let points = traces[i].getElementsByClassName('points');
+            if (points.length !== 0 && points[0].children) {
+                points = points[0].children;
+                for (let j = 0; j < points.length; j++) {
+                    const dimensions = points[j].getBoundingClientRect();
+                    if (evt.event.pageX >= dimensions.left && evt.event.pageX <= dimensions.right
+                        && evt.event.pageY >= dimensions.top && evt.event.pageY <= dimensions.bottom) {
+                        const swapXY = !evt.points[0].data.yaxis;
+                        const dataValue = points[j].__data__.s || (swapXY ? points[j].__data__.x : points[j].__data__.y);
+                        const pointIndex = evt.points.findIndex((trace) => {
+                            if (trace.value) {
+                                return trace.value === dataValue;
+                            }
+                            return swapXY ? trace.x === dataValue : trace.y === dataValue;
+                        });
+                        return evt.points[pointIndex];
+                    }
+                }
+            }
+        }
+        return null;
     };
+    return findPoint(clickEvent, traceDivs);
+}
+/**
+ * Returns array of axis layout keys for the axis that represents
+ * the range of the data
+ *
+ * @param  {Object} layout - Plotly JS layout configuration object.
+ * @return {Array} multiAxes - Array of Plotly JS layout keys
+ */
+function getMultiAxisObjects(layout) { // eslint-disable-line no-unused-vars
+    const multiAxes = [];
+    const layoutKeys = Object.keys(layout);
+    for (let i = 0; i < layoutKeys.length; i++) {
+        if (layout.swapXY) {
+            if (layoutKeys[i].startsWith('xaxis')) {
+                multiAxes.push(layoutKeys[i]);
+            }
+        } else if (layoutKeys[i].startsWith('yaxis')) {
+                multiAxes.push(layoutKeys[i]);
+        }
+    }
+    return multiAxes;
+}
+/**
+ * Determines if legend is located at the top of the chart.
+ *
+ * @param  {Object} layout - Plotly JS layout configuration object.
+ * @return {Boolean} If the legend is top center or not
+ */
+function topLegend(layout) {
```

Comment:
```suggestion
function isTopLegend(layout) {
```
Would also require another change in this file and in `html/.eslintrc.json`.